Parsers/ASimNetworkSession/Parsers/vimNetworkSessionCheckpointFirewall.yaml

```diff
@@ -0,0 +1,169 @@
+Parser:
+  Title: Network Session ASIM parser for Checkpoint Firewall
+  Version: '1.0'
+  LastUpdated: July 15, 2022
+Product:
+  Name: CheckpointFirewall
+Normalization:
+  Schema: NetworkSession
+  Version: '0.2.3'
+References:
+  - Title: ASIM Network Session Schema
+    Link: https://aka.ms/ASimNetworkSessionDoc
+  - Title: ASIM
+    Link: https://aka.ms/AboutASIM
+  - Title: Field Description in Checkpoint Logs 
+    Link: https://supportcenter.checkpoint.com/supportcenter/portal?eventSubmit_doGoviewsolutiondetails=&solutionid=sk144192
+Description: |
+  This ASIM parser supports normalizing Checkpoint VPN-1 & Firewall-1 logs coming from CEF Data Connector to the ASIM Network Session normalized schema.
+ParserName: vimNetworkSessionCheckpointFirewall                               
+EquivalentBuiltInParser: _Im_NetworkSession_CheckpointFirewall                                 
+ParserParams:
+  - Name: starttime
+    Type: datetime
+    Default: datetime(null)
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: srcipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: dstipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: ipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: dstportnumber
+    Type: int
+    Default: int(null)
+  - Name: hostname_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: dvcaction
+    Type: dynamic
+    Default: dynamic([])
+  - Name: eventresult
+    Type: string
+    Default: '*'
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+      let NWParser = (
+          starttime:datetime=datetime(null), 
+          endtime:datetime=datetime(null),
+          srcipaddr_has_any_prefix:dynamic=dynamic([]), 
+          dstipaddr_has_any_prefix:dynamic=dynamic([]), 
+          ipaddr_has_any_prefix:dynamic=dynamic([]),
+          dstportnumber:int=int(null), 
+          hostname_has_any:dynamic=dynamic([]), 
```

Comment:
If there is no hostname info for Checkpoint, you need to check that the value is default (empty dynamic) and fail otherwise. Do it as early as possible after the time filters.